Shared/Samples/List contents of KML file/ListContentsOfKMLFileView.swift

```diff
@@ -0,0 +1,261 @@
+                    try await setUpKMLDataset()
+                } catch {
+                    self.error = error
+                }
+            }
+        }
+        
+        deinit {
+            setupTask?.cancel()
+        }
+        
+        /// Sets up the KML dataset and adds it to the scene as layer.
+        private func setUpKMLDataset() async throws {
+            // Creates the dataset using a local ".kml" file in the bundle.
+            let kmlDataset = KMLDataset(name: "esri_test_data", bundle: .main)!
+            try await kmlDataset.load()
+            self.kmlDataset = kmlDataset
+            
+            // Adds the dataset to the scene as a KML layer.
+            let kmlLayer = KMLLayer(dataset: kmlDataset)
+            scene.addOperationalLayer(kmlLayer)
+            try await scene.load()
+            
+            try await setUpKMLNodes(kmlDataset.rootNodes)
+        }
+        
+        /// Recursively creates viewpoints for KML nodes in a given list.
+        /// - Parameter kmlNodes: The list of KML nodes to set up.
+        private func setUpKMLNodes(_ kmlNodes: [KMLNode]) async throws {
+            // Loads the surface so that the elevation can be queried when the viewpoint is made.
+            try await scene.baseSurface.load()
+            
+            for node in kmlNodes {
+                let viewpoint = try await Viewpoint(kmlNode: node, surface: scene.baseSurface)
+                nodeViewpoints[node.name] = viewpoint
+                
+                // Ensures the node is visible since some are hidden by default.
+                node.isVisible = true
+                
+                if let childNodes = node.children {
+                    try await setUpKMLNodes(childNodes)
+                }
+            }
+        }
+    }
+}
+
+// MARK: Helper Extensions
+
+private extension KMLNode {
+    /// The child nodes of the node, if any.
+    var children: [KMLNode]? {
+        switch self {
+        case let container as KMLContainer:
+            container.childNodes
+        case let networkLink as KMLNetworkLink:
+            networkLink.childNodes
+        default:
+            nil
+        }
+    }
+    
+    /// A human-readable label of the type of the node.
+    var typeLabel: String {
+        switch self {
+        case is KMLDocument: "Document"
+        case is KMLFolder: "Folder"
+        case is KMLContainer: "Container"
+        case is KMLGroundOverlay: "Ground Overlay"
+        case is KMLNetworkLink: "Network Link"
+        case is KMLPhotoOverlay: "Photo Overlay"
+        case is KMLPlacemark: "Placemark"
+        case is KMLScreenOverlay: "Screen Overlay"
+        case is KMLTour: "Tour"
+        default: "Unknown"
+        }
+    }
+}
+
+private extension Viewpoint {
+    /// Creates a viewpoint from a KML node.
+    /// - Parameters:
+    ///   - kmlNode: The KML node.
+    ///   - surface: A surface for determining the elevation needed to offset the viewpoint.
```

Comment:
```suggestion
    ///   - surface: A surface that determines the elevation needed to offset the viewpoint.
```
This doc can be reworded.